server/src/test/scala/io/delta/sharing/server/DeltaSharingServiceSuite.scala

```diff
@@ -1051,7 +1108,7 @@ class DeltaSharingServiceSuite extends FunSuite with BeforeAndAfterAll {
       connection.getInputStream()
     }
     assert(e.getMessage.contains(s"Server returned HTTP response code: $expectedErrorCode"))
-    assert(IOUtils.toString(connection.getErrorStream()).contains(expectedErrorMessage))
+    assert(method == "HEAD" || IOUtils.toString(connection.getErrorStream()).contains(expectedErrorMessage))
```

Comment:
consider adding a comment about why we need the or condition.

Reply:
added.